### PR TITLE
Support ip ranges in endpoint allowedIPAddresses

### PR DESCRIPTION
**Pull request by timokoessler**

### CODE REVIEW

File: aikido_zen/helpers/iplist.py [hunk 1]
```diff
@@ -64,3 +64,9 @@ def matches(self, ip: str) -> bool:
                 return True
 
         return False
+
+    def from_list(self, ip_list):
```
Comment (bitterpanda63) [inline aikido_zen/helpers/iplist.py]: would just move to the constructor right?

File: aikido_zen/background_process/service_config.py [hunk 2]
```diff
@@ -45,10 +45,26 @@ def update(
...
+        # Create a IPList instance for each endpoint
+        for endpoint in self.endpoints:
+            if not "allowedIPAddresses" in endpoint or not isinstance(
+                endpoint["allowedIPAddresses"], list
+            ):
+                #  This feature is not supported by the current aikido server version
+                continue
+            if len(endpoint["allowedIPAddresses"]) == 0:
+                #  Skip empty allowlist
+                continue
```
Comment (bitterpanda63) [inline aikido_zen/background_process/service_config.py:63]: let's add a unit test here for full coverage with empty list

File: aikido_zen/background_process/service_config.py [hunk 3]
```diff
@@ -45,10 +45,28 @@ def update(
...
+            if not "allowedIPAddresses" in endpoint:
+                #  This feature is not supported by the current aikido server version
+                continue
+            if (
+                not isinstance(endpoint["allowedIPAddresses"], list)
+                or len(endpoint["allowedIPAddresses"]) == 0
+            ):
+                #  Skip empty allowlist
+                endpoint["allowedIPAddresses"] = (
+                    None  # Explicitly set to None to be on the safe side
```
Comment (bitterpanda63) [inline aikido_zen/background_process/service_config.py]: not really necessary -> we check type in ip_allowed_to_access_route

**Review by bitterpanda63 — APPROVED** [review]
